test/e2e/xfn_test.go

```diff
@@ -240,3 +240,155 @@ func TestXfnRunnerImagePull(t *testing.T) {
 			Feature(),
 	)
 }
+
+func TestXfnRunnerWriteToTmp(t *testing.T) {
+	manifests := "test/e2e/manifests/xfnrunner/tmp-writer"
+	environment.Test(t,
+		features.New("CreateAFileInTmpFolder").
+			WithLabel(LabelArea, "xfn").
+			WithSetup("InstallRegistry",
+				funcs.AllOf(
+					funcs.AsFeaturesFunc(envfuncs.CreateNamespace("reg")),
+					funcs.AsFeaturesFunc(
+						funcs.HelmRepo(
+							helm.WithArgs("add"),
+							helm.WithArgs("twuni"),
+							helm.WithArgs("https://helm.twun.io"),
+						)),
+					funcs.AsFeaturesFunc(
+						funcs.HelmInstall(
+							helm.WithName("public"),
+							helm.WithNamespace("reg"),
+							helm.WithWait(),
+							helm.WithChart("twuni/docker-registry"),
+							helm.WithVersion("2.2.2"),
+							helm.WithArgs(
+								"--set service.type=NodePort",
+								"--set service.nodePort=32000",
+							),
+						))),
+			).
+			WithSetup("CopyFnImageToRegistry", func(ctx context.Context, t *testing.T, config *envconf.Config) context.Context {
```

Comment:
Given that this is mostly the same as the one used above, it could be worth factoring out this to a dedicated helper function in funcs, something like:
```
func CopyImageToRegistry(image, dstRegistry string, timeout time.Duration) features.Func {
	return func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
		...
	}
}
```
something like

Reply:
There are lots of common steps in the two e2e tests. I'll refactor the code once I verified that the test is working :)

Reply:
Awesome thanks 🙏